### ISSUE TITLE
LaTex package newunicodechar: "package won't work "

### ISSUE THREAD

**Issue by algo99** [doxygen/doxygen#6775]
Trying to use plaintext Unicode (UTF-8-encoded) in a documentation  of a source code in doxygen output get this warning:
> Package newunicodechar Warning: This package won't work without loading
> (newunicodechar)                `inputenc' or `inputenx' with the `utf8' option

Looks like the LaTex option
`\usepackage[utf8]{inputenc}`
is missing.

**Comment by albert-github** [comment]
Which version of doxygen are you using?
Which LaTeX distribution are you using (see e.g. `pdflatex --version`)

Does the message also appear when you use the current (1.8.15) doxygen version?

**Comment by algo99** [comment]
> Which version of doxygen are you using?

master

> Which LaTeX distribution are you using (see e.g. `pdflatex --version`)

$ pdflatex --version
pdfTeX 3.14159265-2.6-1.40.15 (TeX Live 2014)
kpathsea version 6.2.0
Copyright 2014 Peter Breitenlohner (eTeX)/Han The Thanh (pdfTeX).
There is NO warranty.  Redistribution of this software is
covered by the terms of both the pdfTeX copyright and
the Lesser GNU General Public License.
For more information about these matters, see the file
named COPYING and the pdfTeX source.
Primary author of pdfTeX: Peter Breitenlohner (eTeX)/Han The Thanh (pdfTeX).
Compiled with libpng 1.5.13; using libpng 1.5.30
Compiled with zlib 1.2.7; using zlib 1.2.7
Compiled with poppler version 0.26.5

> 
> Does the message also appear when you use the current (1.8.15) doxygen version?

Just tested, the warning goes away if use this fix
```
$ git diff
diff --git a/src/util.cpp b/src/util.cpp
index 9a0e513b..fc3dad52 100644
--- a/src/util.cpp
+++ b/src/util.cpp
@@ -9021,7 +9021,8 @@ void writeLatexSpecialFormulaChars(FTextStream &t)
     sup3[1]= 0xB3;
     sup3[2]= 0;
 
-    t << "\\usepackage{newunicodechar}\n"
+    t << "\\usepackage[utf8]{inputenc}\n"
+         "\\usepackage{newunicodechar}\n"
          "  \\newunicodechar{" << pminus << "}{${}^{-}$}% Superscript minus\n"
          "  \\newunicodechar{" << psup2  << "}{${}^{2}$}% Superscript two\n"
          "  \\newunicodechar{" << psup3  << "}{${}^{3}$}% Superscript three\n"
```

**Comment by albert-github** [comment]
Looks like this is something in the formulas are you use any specific language settings?
I didn't observe the problem.

Can you please attach a, small, self contained example (source+config file in a tar or zip) that allows us to reproduce the problem?

**Comment by algo99** [comment]
We have a large codebase and I could not figure out which of our source files triggers pdflatex call. Look like pdflatex is started from doxygen in a separate thread and looking through doxygen output alone can not help. 

So I tried first to find out what else Unicode characters we have. These are:
* degree symbol
* german umlauts

**Comment by albert-github** [comment]
Problem might be with the input encoding as well, like requested before a small example could shine a light on it.

**Comment by algo99** [comment]
I have managed to prepare an example.

Launching doxygen-master with this configuration:
```
# Doxyfile 1.8.15
JAVADOC_AUTOBRIEF      = YES
FILE_PATTERNS          = *.hpp
RECURSIVE              = YES
GENERATE_LATEX         = NO
```

for this C++ header:
```
/**
 * be the average distance between two points (default = @f$20cm@f$).
 */
const double DEFAULT_PANNING_DISTANCE = 0.2;
```
gives this warning:

<details><summary>Click to expand</summary><p>

```
...
Generating bitmaps for formulas in HTML...
This is pdfTeX, Version 3.14159265-2.6-1.40.18 (TeX Live 2017) (preloaded format=latex)
 restricted \write18 enabled.
entering extended mode
(./_formulas.tex
LaTeX2e <2017-04-15>
Babel <3.9r> and hyphenation patterns for 61 language(s) loaded.
(/usr/share/texlive/texmf-dist/tex/latex/base/article.cls
Document Class: article 2014/09/29 v1.4h Standard LaTeX document class
(/usr/share/texlive/texmf-dist/tex/latex/base/size10.clo))
(/usr/share/texlive/texmf-dist/tex/latex/base/ifthen.sty)
(/usr/share/texlive/texmf-dist/tex/latex/graphics/epsfig.sty
(/usr/share/texlive/texmf-dist/tex/latex/graphics/graphicx.sty
(/usr/share/texlive/texmf-dist/tex/latex/graphics/keyval.sty)
(/usr/share/texlive/texmf-dist/tex/latex/graphics/graphics.sty
(/usr/share/texlive/texmf-dist/tex/latex/graphics/trig.sty)
(/usr/share/texlive/texmf-dist/tex/latex/graphics-cfg/graphics.cfg)
(/usr/share/texlive/texmf-dist/tex/latex/graphics/dvips.def))))
(/usr/share/texlive/texmf-dist/tex/latex/newunicodechar/newunicodechar.sty

Package newunicodechar Warning: This package won't work without loading
(newunicodechar)                `inputenc' or `inputenx' with the `utf8' option
.

)
No file _formulas.aux.
[1] (./_formulas.aux) )
Output written on _formulas.dvi (1 page, 252 bytes).
Transcript written on _formulas.log.
Generating image form_0.png for formula
...
```

</p></details>

**Comment by albert-github** [comment]
Even with your example I was not able to reproduce the problem though I'm using "This is pdfTeX, Version 3.14159265-2.6-1.40.19 (TeX Live 2018/Cygwin)" and "This is pdfTeX, Version 3.14159265-2.6-1.40.19 (MiKTeX 2.9.6840 64-bit)", so both just a bit newer than the version you are using but it should work with your version as well.

To be sure that there is nothing special in the input files, please attach a, small, self contained example (source+config file in a tar or zip) that allows us to reproduce the problem? In this case the `.hpp` and `Doxyfile` should be sufficient.

**Comment by algo99** [comment]
Actually these files have the same content as in my previous comment.
[6775.zip](https://github.com/doxygen/doxygen/files/2775985/6775.zip)

Maybe the resulting warning depends on the version of newunicodechar package?

Looking here: https://ctan.org/tex-archive/macros/latex/contrib/newunicodechar
> This is version 1.2 of the package
>
> Changes from version 1.1:
>  o Now utf8 is default for 8 bit engines

Till today I tested doxygen on Fedora 28 which has 1.1 version (so you have to specify the encoding) of newunicodechar package
```shell
$ rpm -qa | grep newunicodechar
texlive-newunicodechar-svn28253.1.1-52.fc28.2.noarch
```
Today I tried the above example on Windows with TeX Live 2018 (which has newunicodechar-1.2) and the warning  was gone!

**Comment by albert-github** [comment]
Thanks a lot for this valuable research!

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #6785 (it doesn't look like to have a negative effect in case the option is already theer by default)

As a workaround it will probably be sufficient to define an own style file (e.g. my_formula_style.sty) with just one line in it :

```\usepackage[utf8]{inputenc}```

and add this style file to the configuration tag `LATEX_EXTRA_STYLESHEET`

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of a release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.16.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
